server/repository/src/migrations/v2_03_00/add_vaccinations_table.rs

```diff
@@ -0,0 +1,36 @@
+use crate::migrations::*;
+
+pub(crate) struct Migrate;
+
+impl MigrationFragment for Migrate {
+    fn identifier(&self) -> &'static str {
+        "add_vaccinations_table"
+    }
+
+    fn migrate(&self, connection: &StorageConnection) -> anyhow::Result<()> {
+        // Note: some fields don't have reference constraints, as those records may not be synced to Central OMS
+        // or to other facilities viewing the vaccination records
+        sql!(
+            connection,
+            r#"
+                CREATE TABLE vaccination (
+                    id TEXT NOT NULL PRIMARY KEY,
+                    program_id TEXT NOT NULL,
+                    encounter_id TEXT NOT NULL,
+                    created_datetime TIMESTAMP NOT NULL,
+                    user_id TEXT NOT NULL REFERENCES user_account(id),
+                    vaccine_course_dose_id TEXT NOT NULL REFERENCES vaccine_course_dose(id),
```

Comment:
Any plans to support vaccinations that aren't part of a course? E.g. Influenza?
Just wondering if `Not Null` is too strong constraint here...

Reply:
Great question... do you have preferences/upcoming requirements here @mark-prins?

Reply:
No requirement at this stage to be able to administer items that aren't part of the course.
Doing so wouldn't fit into the concept of a vaccination card; which is to define a course of vaccines which are then ticked off / administered according to the schedule. So I'm fine with keeping the course_dose_id a hard link